server/tests/backends/sql_translator/common.py

```diff
@@ -0,0 +1,42 @@
+import math
+from contextlib import suppress
+from decimal import Decimal
+from typing import Any
+
+import pandas as pd
+
+
+def standardized_columns(df: pd.DataFrame, colname_lowercase: bool = False):
+
+    df.columns = [
+        (c.replace('-', '_').lower() if colname_lowercase else c.replace('-', '_'))
+        for c in df.columns
+    ]
+
+
+def standardized_values(df: pd.DataFrame, convert_nan_to_none: bool = False) -> None:
+    # We can standardized the floating point
+
+    def _clean_float(first_value: Any, colname: str) -> None:
+        with suppress(Exception):
+            for i, v in enumerate(df[colname].values):
+                # we truncate the decimal part to have only 10 values (withouth
+                # rounding it)
+                df[colname].values[i] = math.floor(v * 10**10) / 10**10
+
+    for colname in df:
+        if df[colname].dtype == 'object':
+            # get the first non-null value in the series.
+            # if it's a float, try to convert the whole serie with the
+            # precision we want.
+            # If it's string and == to 'nan', convert that to None/null
+            s = df[colname]
+            idx = s.first_valid_index()
+            if idx is not None and (first_value := s.loc[idx]) is not None:
+                if isinstance(first_value, (Decimal, float)) and "." in str(first_value):
+                    _clean_float(first_value, colname)
+
+                if convert_nan_to_none:
+                    for i, v in enumerate(df[colname].values):
+                        if str(df[colname].values[i]).lower() == 'nan':
+                            df[colname].values[i] = None
```

Comment:
These functions seem dangerous to me, as we're applying some transformation to data that won't be applied when a pipeline is actually executed. We can keep `standardized_columns` and the  `_clean_float` part of `standardized_values` for now until postgres pypika tests are ISO with the rest, but I'm against the nan to None part for the following reasons:

1. The behaviour is undefined, depending on the pandas version: Either the series' type will change from `float` to object, or the `None` will be converted to `nan` again
2. `assert_dataframes_equal` should treat all null-ish values as equal (basically if `pd.isnull(v1) and pd.isnull(v2)` we can consider `v1 == v2`)

Reply:
Alright, you got a point !
I removed that here : https://github.com/ToucanToco/weaverbird/pull/1390/commits/cdd2c0e860c23fae385fd1fbc0ffe70ea3588adf

But i had to skip the convert_text step since it was failing : 

```
>   ???
E   AssertionError: DataFrame.iloc[:, 0] (column name="value") are different
E
E   DataFrame.iloc[:, 0] (column name="value") values are different (16.66667 %)
E   [index]: [0, 1, 2, 3, 4, 5]
E   [left]:  [41, 42, 43.5, 43.6, nan, meh]
E   [right]: [41, 42, 43.5, 43.6, None, meh]

pandas/_libs/testing.pyx:167: AssertionError
================================================== short test summary info ===================================================
FAILED tests/backends/sql_translator_integration_tests/test_sql_postgres_pypika_translator_steps.py::test_sql_translator_pipeline[/convert_to_text]
```

Reply:
Seems legit. We'd have to agree on a convention between all translators to decide what the value for null should be as text. Should it be an emtpy string, the litteral `"null"` something else... ? But that's out of the scope of this PR